### PR TITLE
aldente: init at 1.22

### PR DESCRIPTION
**Pull request by stepbrobd**
###### Description of changes

add new darwin only package

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by StepBroBD** [comment]
On aarch64-darwin:
```shell
$ nix run nixpkgs#nixpkgs-review -- pr 242000
1 package added:
aldente (init at 1.22)
1 package built:
aldente
```

File: pkgs/os-specific/darwin/aldente/default.nix [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      rm -rf $mnt
+    }
+    trap finish EXIT
+
+    echo "Attaching $mnt"
+    /usr/bin/hdiutil attach -nobrowse -readonly $src -mountpoint $mnt
+
+    echo "What's in the mount dir"?
+    ls -la $mnt/
+
```
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/aldente/default.nix:43]:
```suggestion
```
debugging code should be removed

File: pkgs/os-specific/darwin/aldente/default.nix [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    echo "Copying contents"
```
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/aldente/default.nix:44]:
```suggestion
```
phases are already announced and if things happen in the right phase there shouldn't be confusion

File: pkgs/os-specific/darwin/aldente/default.nix [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+    function finish {
+      echo "Detaching $mnt"
+      /usr/bin/hdiutil detach $mnt -force
+      rm -rf $mnt
+    }
+    trap finish EXIT
+
+    echo "Attaching $mnt"
+    /usr/bin/hdiutil attach -nobrowse -readonly $src -mountpoint $mnt
```
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/aldente/default.nix:39]:
```suggestion
    if ! [[ "$curSrc" =~ \.dmg$ ]]; then return 1; fi
    mnt=$(mktemp -d -t ci-XXXXXXXXXX)

    function finish {
      /usr/bin/hdiutil detach $mnt -force
    }
    trap finish EXIT

    /usr/bin/hdiutil attach -nobrowse -readonly $src -mountpoint $mnt
```

do we really need the trap? Is this not happening inside the sandbox?!

File: pkgs/os-specific/darwin/aldente/default.nix [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    name = "AlDente.dmg";
+    url = "https://github.com/davidwernhart/aldente-charge-limiter/releases/download/${finalAttrs.version}/AlDente.dmg";
+    hash = "sha256-y8S9SCDbvoKGtNqGQvJDSpbLhh7GqXUo7pLcMzuCnjo=";
+  };
+
+  dontPatch = true;
+  dontConfigure = true;
+  dontBuild = true;
+  dontFixup = true;
+
```
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/aldente/default.nix:21]:
```suggestion
  dontBuild = true;
  dontFixup = true;

```
patching and configure should be no-ops. Build maybe, too and is fixing up breaking things?

File: pkgs/os-specific/darwin/aldente/default.nix [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+, fetchurl
+, undmg
+}:
+
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "aldente";
+  version = "1.22";
+
+  src = fetchurl {
+    name = "AlDente.dmg";
```
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/aldente/default.nix:12]:
```suggestion
```

**Comment by StepBroBD** [comment]
Thanks for the review! I committed your requested changes and squashed them.

On aarch64-darwin:

```shell
$ nix run nixpkgs#nixpkgs-review -- pr 242000
1 package added:
aldente (init at 1.22)
1 package built:
aldente
```

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Pull request merged**